baseplate/sidecars/live_data_watcher.py

```diff
@@ -85,7 +85,13 @@ def get_data_to_write(json_data: dict, data: Optional[bytes]) -> Optional[bytes]
         # }
         # If the load type is 'S3', this format is an indication that we support
         # downloading the contents of encrypted files uploaded to S3.
-        if json_data.get("live_data_watcher_load_type") == "S3":
+        is_s3 = False
+        try:
+            is_s3 = json_data["live_data_watcher_load_type"] == "S3"
+        except (KeyError, TypeError):
```

Comment:
since it's one line it's fine but I probably would have opted for `is_s3 = json_data.get("live_data_watcher_load_type", "zk") == "S3"` to avoid the KeyError

Reply:
The type for json_data on line 81 should be updated to indicate it can be more than a dict.

Reply:
I agree with @nsheaps original comment about using `.get(<key>, <default>)`.


I think we might need to catch `AttributeError` as well, if the data happens to be formatted as a json list, that will parse as JSON but it will be a list not an object, like this:

```
>>> import json
>>> raw_json="""
... ["foo", "bar"]
... """
>>> parsed_json=json.
>>> parsed_json=json.loads(raw_json)
>>> parse_json.get("foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'parse_json' is not defined
>>> parsed_json.get("foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list' object has no attribute 'get'
```

Reply:
Thanks for looking/checking the exception type @KTAtkinson